File: 2283/2283.cs
public class Solution {
    public int CheckIfNumberHasEqualDigitCoun(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(CheckIfNumberHasEqualDigitCoun(root.left), CheckIfNumberHasEqualDigitCoun(root.right));
    }
}

File: 2283/2283.go
func checkIfNumberHasEqualDigitCoun(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := checkIfNumberHasEqualDigitCoun(root.Left)
	r := checkIfNumberHasEqualDigitCoun(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}